server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSPerformSPOptionsCheckUtility.java

```diff
@@ -0,0 +1,282 @@
+                checkSsl((CICSParser.Cics_perform_sslContext) ctx);
+                break;
+            case RULE_cics_perform_statistics:
+                checkstatistics((CICSParser.Cics_perform_statisticsContext) ctx);
+                break;
+            default:
+                break;
+        }
+        checkDuplicates(ctx);
+    }
+    private void checkDump(CICSParser.Cics_perform_dumpContext ctx) {
+        checkHasMandatoryOptions(ctx.DUMP(), ctx, "DUMP");
+        checkHasMandatoryOptions(ctx.DUMPCODE(), ctx, "DUMPCODE");
+        if (!ctx.TITLE().isEmpty() || !ctx.TITLELENGTH().isEmpty()) {
+            checkHasMandatoryOptions(ctx.TITLE(), ctx, "TITLE");
+            checkHasMandatoryOptions(ctx.TITLELENGTH(), ctx, "TITLELENGTH");
+        }
+        if (!ctx.CALLER().isEmpty() || !ctx.CALLERLENGTH().isEmpty()) {
+            checkHasMandatoryOptions(ctx.CALLER(), ctx, "CALLER");
+            checkHasMandatoryOptions(ctx.CALLERLENGTH(), ctx, "CALLERLENGTH");
+        }
+    }
+    private void checkEndAffinity(CICSParser.Cics_perform_endaffinityContext ctx) {
+        checkHasMandatoryOptions(ctx.ENDAFFINITY(), ctx, "ENDAFFINITY");
+        checkHasMandatoryOptions(ctx.NETNAME(), ctx, "NETNAME");
+    }
+    private void checkJvmServer(CICSParser.Cics_perform_jvmserverContext ctx) {
+        checkHasMandatoryOptions(ctx.JVMSERVER(), ctx, "JVMSERVER");
+        checkHasExactlyOneOption("JVMTYPE or JVM or LIBERTY or OSGI", ctx, ctx.JVMTYPE(), ctx.JVM(), ctx.LIBERTY(), ctx.OSGI());
+        if (!ctx.JVM().isEmpty()) {
+            checkHasExactlyOneOption("JVMACTION or DUMP or GATHER or STACKTRACE", ctx, ctx.JVMACTION(), ctx.DUMP(), ctx.GATHER(), ctx.STACKTRACE());
+            if (!ctx.DUMP().isEmpty()) {
+                checkHasExactlyOneOption("DUMPTYPE or ALL or JAVACORE or HEAP or SNAPTRACE", ctx, ctx.DUMPTYPE(), ctx.ALL(), ctx.JAVACORE(), ctx.HEAP(), ctx.SNAPTRACE());
+            } else if (!ctx.GATHER().isEmpty()) {
+                checkHasExactlyOneOption("GATHERTYPE or DIAGNOSTICS", ctx, ctx.GATHERTYPE(), ctx.DIAGNOSTICS());
+            } else if (!ctx.STACKTRACE().isEmpty() || !ctx.TASKID().isEmpty()) {
+                checkHasMandatoryOptions(ctx.STACKTRACE(), ctx, "STACKTRACE");
+                checkHasMandatoryOptions(ctx.TASKID(), ctx, "TASKID");
+            }
+        } else if (!ctx.LIBERTY().isEmpty()) {
+            checkHasExactlyOneOption("LIBRTYACTION or REFRESH or SERVERDUMP", ctx, ctx.LIBRTYACTION(), ctx.REFRESH(), ctx.SERVERDUMP());
+            if (!ctx.REFRESH().isEmpty()) {
+                checkHasExactlyOneOption("RESOURCETYPE or APPLICATION or CONFIG", ctx, ctx.RESOURCETYPE(), ctx.APPLICATION(), ctx.CONFIG());
+                if (!ctx.APPLICATION().isEmpty() || !ctx.APPID().isEmpty()) {
+                    checkHasMandatoryOptions(ctx.APPLICATION(), ctx, "APPLICATION");
+                    checkHasMandatoryOptions(ctx.APPID(), ctx, "APPID");
+                }
```

Comment:
```suggestion
                if (!ctx.APPLICATION().isEmpty()) {
                    checkHasMandatoryOptions(ctx.APPID(), ctx, "APPID");
                }
```
or checkPrerequisiteIsMet